component/module/git/internal/vcs/auth/auth.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/grafana/agent/pkg/river/rivertypes"
+)
+
+type GitAuthConfig struct {
+	BasicAuth *BasicAuth `river:"basic_auth,block,optional"`
+	SSHKey    *SSHKey    `river:"ssh_key,block,optional"`
+}
+
+// Convert converts HTTPClientConfig to the native Prometheus type. If h is
+// nil, the default client config is returned.
+func (h *GitAuthConfig) Convert() transport.AuthMethod {
+	if h == nil {
+		return nil
+	}
+
+	if h.BasicAuth != nil {
+		return h.BasicAuth.Convert()
+	}
+
+	if h.SSHKey != nil {
+		key, _ := h.SSHKey.Convert()
+		return key
+	}
+	return nil
+}
+
+type BasicAuth struct {
+	Username string            `river:"username,attr,optional"`
+	Password rivertypes.Secret `river:"password,attr,optional"`
+}
```

Comment:
I don't think these are both optional; shouldn't they both be required for basic auth?

Reply:
That's a great question. I thought the same thing and I'm happy to make that change, I did just copy this logic from https://github.com/grafana/agent/blob/main/component/common/config/types.go#L130

Reply:
Interesting, we probably had that discussion before and it turned out they truly should be optional. This can stay as-is, then 👍